articles/agora-express-token-server/index.md

```diff
@@ -0,0 +1,364 @@
+  const appCertificate = "521c7307f1d547e080c34585dd20662e";
+  const expirationTimeInSeconds = 3600;
+  const uid = Math.floor(Math.random() * 100000);
+  const role = req.body.isPublisher ? Agora.RtcRole.PUBLISHER : Agora.RtcRole.SUBSCRIBER;
+  const channel = req.body.channel;
+  const currentTimestamp = Math.floor(Date.now() / 1000);
+  const expirationTimestamp = currentTimestamp + expirationTimeInSeconds;
+
+  const token = Agora.RtcTokenBuilder.buildTokenWithUid(appID, appCertificate, channel, uid, role, expirationTimestamp);
+  res.send({ uid, token });
+});
+```
+
+Request & Response: 
+
+![RTC Token Request and Response](rtc_postman.png)
+
+### RTM Token
+Let's add a POST handler for a new endpoint called `'/rtmtoken'` to generate authentication tokens for the RTM SDK.
+
+```JavaScript
+app.post("/rtmtoken", (req, res) => {
+  // Generate Token Here
+});
+```
+
+When someone requests this endpoint, we'll generate an RTM authentication token and send it as the response.
+
+The `agora-access-token` provides a class called `RtmTokenBuilder` which has a method called `buildToken`.
+
+The `buildToken` method requires 5 arguments.
+
+1. App ID
+2. App Certificate
+3. User Account
+4. User Role
+5. Expiration Time
+
+We have the app ID and the App Certificate from the project management console.
+
+```JavaScript
+const appID = "<-- Your app ID Here -->";
+const appCertificate = "<-- Your App Certificate Here -->";
+```
+
+We should get the user account from the request.
+
+```JavaScript
+const user = req.body.user;
+```
+
+Let's set the expiration time of the token. You need to provide a timestamp for the expiration time. We'll add the time (in seconds) to the current timestamp to set when we want the token to expire.
+
+```JavaScript
+const expirationTimeInSeconds = 3600;
+const currentTimestamp = Math.floor(Date.now() / 1000);
+const privilegeExpiredTs = currentTimestamp + expirationTimeInSeconds;
+```
+
+There's only one role in RTM, which is an RTM User.
+
+```JavaScript
+const role = Agora.RtmRole.Rtm_User;
+```
+
+Let's pass these data to `buildToken` to generate the token.
+
+```JavaScript
+const token = Agora.RtmTokenBuilder.buildToken(appID, appCertificate, uid, role, privilegeExpiredTs);
+```
+
+Now that we have the token, Let's return the token as the response.
+
+```JavaScript
+res.send({ token });
+```
+
+Final Code:
+
+```JavaScript
+app.post("/rtmtoken", (req, res) => {
+  const appID = "e004df68b67841a1b6ea7c08b7225814";
+  const appCertificate = "521c7307f1d547e080c34585dd20662e";
+  const user = req.body.user;
+  const role = Agora.RtmRole.Rtm_User;
+  const expirationTimeInSeconds = 3600;
+  const currentTimestamp = Math.floor(Date.now() / 1000);
+  const expirationTimestamp = currentTimestamp + expirationTimeInSeconds;
+
+  const token = Agora.RtmTokenBuilder.buildToken(appID, appCertificate, user, role, expirationTimestamp);
+  res.send({ token });
+});
+```
+
+Request & Response: 
```

Comment:
Consider adding a code fence with the post body and making it clear that one has to use the "uid" value returned by the rtctoken request and append "user_" before it.

Reply:
I've added a documentation link for User Accounts. They are basically UID but in string format. There's a little bit more procedure on the Front end side, that doesn't concern the token generation.

Reply:
Oh okay then